src/GridFS/ReadableStream.php

```diff
@@ -250,16 +255,25 @@ private function initBufferFromCurrentChunk(): bool
             return false;
         }
 
+        if ($this->chunksIterator === null) {
+            return false;
+        }
+
         if (! $this->chunksIterator->valid()) {
             throw CorruptFileException::missingChunk($this->chunkOffset);
         }
 
         $currentChunk = $this->chunksIterator->current();
+        assert(is_object($currentChunk));
 
         if ($currentChunk->n !== $this->chunkOffset) {
             throw CorruptFileException::unexpectedIndex($currentChunk->n, $this->chunkOffset);
         }
 
+        if (! $currentChunk->data instanceof Binary) {
+            throw CorruptFileException::invalidChunkData($this->chunkOffset);
+        }
```

Comment:
This seems like a reasonable check that would not have been gracefully handled previously. It's quite possible someone could corrupt the `data` field in a chunk document since MongoDB doesn't enforce a schema on that collection.